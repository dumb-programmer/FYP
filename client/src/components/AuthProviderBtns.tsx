import googleIcon from "@/assets/google-icon.svg";
import githubIcon from "@/assets/github-icon.svg";

export default function AuthProviderBtns() {
    return <div className="flex flex-col gap-4" data-testid="auth-provider-btns">
        <a href="http://localhost:3000/auth/google" className="btn glass flex items-center gap-4">
            <img src={googleIcon} style={{ transform: "scale(1.3)" }} alt="google" />
            Continue With Google
        </a>
        <a href="http://localhost:3000/auth/github" className="btn glass flex items-center gap-4">
            <img src={githubIcon} style={{ transform: "scale(1.8)" }} alt="github" />
            Continue With Github
        </a>
    </div>;
}