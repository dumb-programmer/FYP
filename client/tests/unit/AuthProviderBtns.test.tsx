import React from "react";
import { render, screen } from "@testing-library/react";
import { expect, describe, it } from "vitest";
import AuthProviderBtns from "../../src/components/AuthProviderBtns";

const renderComponent = () => render(<AuthProviderBtns />);

describe("AuthProviderBtns", () => {
    it("renders correctly", () => {
        renderComponent();

        const authProviderBtns = screen.getByTestId("auth-provider-btns");

        expect(authProviderBtns).toMatchSnapshot();
    });

    it("continue with google button has redirect link to google provider", () => {
        renderComponent();

        const continueWithGoogleButton = screen.getByRole("link", { name: /continue with google/i });

        expect(continueWithGoogleButton).toHaveAttribute("href", "http://localhost:3000/auth/google");
    });

    it("continue with github button has redirect link to github provider", () => {
        renderComponent();

        const continueWithGithubButton = screen.getByRole("link", { name: /continue with github/i });

        expect(continueWithGithubButton).toHaveAttribute("href", "http://localhost:3000/auth/github");
    });
})