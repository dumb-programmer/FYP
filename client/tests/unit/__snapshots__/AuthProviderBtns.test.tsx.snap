// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthProviderBtns > renders correctly 1`] = `
<div
  class="flex flex-col gap-4"
  data-testid="auth-provider-btns"
>
  <a
    class="btn glass flex items-center gap-4"
    href="http://localhost:3000/auth/google"
  >
    <img
      alt="google"
      src="/src/assets/google-icon.svg"
      style="transform: scale(1.3);"
    />
    Continue With Google
  </a>
  <a
    class="btn glass flex items-center gap-4"
    href="http://localhost:3000/auth/github"
  >
    <img
      alt="github"
      src="/src/assets/github-icon.svg"
      style="transform: scale(1.8);"
    />
    Continue With Github
  </a>
</div>
`;
